File: app/src/main/java/com/udacity/maluleque/meutako/utils/NetworkServiceChecker.java
package com.udacity.maluleque.meutako.utils;

import android.content.Context;
import android.os.AsyncTask;
import android.util.Log;

public class NetworkServiceChecker extends AsyncTask<Void, Void, Boolean> {

    private static final String TAG = "NetworkServiceChecker";
    NetworkNotifier networkNotifier;
    Context context;

    public NetworkServiceChecker(Context context, NetworkNotifier networkNotifier) {
        this.context = context;
        this.networkNotifier = networkNotifier;
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
        Log.d(TAG, "onPreExecute starting to check internet connection");
    }

    @Override
    protected Boolean doInBackground(Void... voids) {
        return NetworkUtils.hasInternetConnection(context);
    }

    protected void onPostExecute(Boolean hasActiveConnection) {
        networkNotifier.notifyInternetConnection(hasActiveConnection);
        Log.d(TAG, "onPostExecute hasConnection: " + hasActiveConnection);
    }
}

File: app/src/main/java/com/udacity/maluleque/meutako/utils/NetworkServiceChecker.kt
package com.udacity.maluleque.meutako.utils

import android.content.Context
import android.os.AsyncTask
import android.util.Log

class NetworkServiceChecker(var context: Context, var networkNotifier: NetworkNotifier) : AsyncTask<Void?, Void?, Boolean>() {

    override fun onPreExecute() {
        super.onPreExecute()
        Log.d(TAG, "onPreExecute starting to check internet connection")
    }

    override fun doInBackground(vararg voids: Void?): Boolean {
        return NetworkUtils.hasInternetConnection(context)
    }

    override fun onPostExecute(hasActiveConnection: Boolean) {
        networkNotifier.notifyInternetConnection(hasActiveConnection)
        Log.d(TAG, "onPostExecute hasConnection: $hasActiveConnection")
    }

    companion object {
        private const val TAG = "NetworkServiceChecker"
    }


}